### ISSUE TITLE
Bazel CI Release Process should build and publish Bazel docker containers

### ISSUE THREAD

**Issue by philwo** [bazelbuild/continuous-integration#1060]
The current Bazel Docker containers are being built by the RBE toolchains team, but they asked us to take this on.

We should setup some tooling and a pipeline (?) to regularly automatically build current Bazel containers for all released versions.

**Comment by kevingessner** [comment]
@philwo Hello, any news here? This is still blocking us on smoothly upgrading Bazel past 3.5.

**Comment by philwo** [comment]
@kevingessner Not yet, sorry :( But this is really important to me and I'll get to it as soon as I can, promised!

**Comment by EdSchouten** [comment]
Any updates on this?

**Comment by chasain** [comment]
I'm all for automating this, but for now can we get a manually built image pushed

**Comment by johanbrandhorst** [comment]
Friendly bump, blocking upgrade from 3.5.0

**Comment by EdSchouten** [comment]
This is still an issue.

**Comment by lbhdc** [comment]
I came across this issue while searching for a docker image that supported bazel 4. I didn't find any, so I made my own and thought I would share.

```dockerfile
# Dockerfile
FROM registry.fedoraproject.org/fedora:34

RUN dnf install -y dnf-plugins-core git patch python3 && \
    dnf copr enable -y vbatts/bazel                   && \
    dnf install -y bazel4                             && \
    dnf clean all                                     && \
    rm -rf /var/cache/yum

RUN ln -s /usr/bin/python3 /usr/bin/python

WORKDIR /home/builder

ENTRYPOINT ["bazel"]
```

```shell
$ docker build .
...
Successfully built <container-id>
$ docker run <container-id> --version
bazel 4.2.1
```

**Comment by dayfine** [comment]
It has been almost a year. Seems a bit long for a P1? Should it be now bumped to P0 to help prioritization?

I came here after updating `rules_python` to 0.5.0. It mentions Bazel 4 is now the LTS version. I suppose P0 might make sense?

**Comment by ScarletFlash** [comment]
Any updates here?

**Comment by QuantamHD** [comment]
@ScarletFlash I'm not on or affiliated with the Bazel team, but I reached out to one of my contacts inside Google to help get this prioritized.

**Comment by philwo** [comment]
I talked with @alexeagle about this last week who wanted to help me out (thanks!!). Alex, do you already have a Dockerfile that I could try?

**Comment by f0rmiga** [comment]
@philwo Alex asked if I could come up with a Dockerfile for this. How about https://github.com/bazelbuild/continuous-integration/pull/1290?

**Comment by davidvandebunte** [comment]
According to [this thread](https://groups.google.com/g/bazel-dev/c/JausNtNZBGo) the current bazel image is based on the [GoogleCloudPlatform/container-definitions](https://github.com/GoogleCloudPlatform/container-definitions) repository. Can someone double confirm that is the case? Perhaps this is related to @philwo's initial comment on this thread about the RBE toolchains team giving up control. I'm simply trying to find the code that was used to build the current image.

**Comment by iangudger** [comment]
Maybe this could be manually updated for Bazel 5.0?

**Comment by abhillman** [comment]
Potential workaround is to use the cockroach db bazel images https://hub.docker.com/r/cockroachdb/bazel/tags

**Comment by dlorenc** [comment]
Hey - we have bazel images published by Chainguard here at cgr.dev/chainguard/bazel! Feel free to try them out.

**Comment by meteorcloudy** [comment]
I'm closing this one since bazel docker images are already published to gcr.io/bazel-public/bazel for every new Bazel version

**Comment by yogurtearl** [comment]
> I'm closing this one since bazel docker images are already published to gcr.io/bazel-public/bazel for every new Bazel version

@meteorcloudy could/should that info be added to the bazel docs? it is not very discoverable.

**Comment by meteorcloudy** [comment]
@yogurtearl SGTM, sent https://github.com/bazelbuild/bazel/pull/17113

We also have documentation in our release process: https://github.com/bazelbuild/continuous-integration/blob/master/docs/release-playbook.md#push-new-docker-image

**Comment by loeffel-io** [comment]
Hey @meteorcloudy, 

we are using the gcr.io/bazel-public/bazel images since years, thank you very much btw. 

Is there any way to run the images as nonroot? ref: https://github.com/bazelbuild/rules_python/issues/1169

**Comment by meteorcloudy** [comment]
@loeffel-io I guess you can just create and switch to a non-root user? Also the rules_python problem is fixed in https://github.com/bazelbuild/rules_python/pull/2636

**Comment by loeffel-io** [comment]
> [@loeffel-io](https://github.com/loeffel-io) I guess you can just create and switch to a non-root user? Also the rules_python problem is fixed in [bazelbuild/rules_python#2636](https://github.com/bazelbuild/rules_python/pull/2636)

Thanks @meteorcloudy, just found the Dockerfile. 

Yes but until all deps are up 2 date will take some time. Wouldn't it make sense to release non root tags like distroless for the bazel images? not sure if bazel requires root access.

btw i always thought that the bazel docker image is based on distroless instead of ubuntu for hermetic purposes
